Sources/main.cpp

```diff
@@ -355,8 +365,12 @@ static int GdbserverMain(int argc, char **argv) {
   else
     impl = ds2::make_unique<DebugSessionImpl>();
 
-  return RunDebugServer(reverse ? socket.get() : socket->accept().get(),
-                        impl.get());
+  if (fd >= 0) {
+    return RunDebugServer(socket.get(), impl.get());
```

Comment:
It'd be nice if we could avoid having 2 return statements. As in, `RunDEbugServer` takes 2 argument and the second one is the same each time. It'd be nice if we could use this if/else to choose just the first argument and call `RunDebugServer` under it.

But do we actually need to have this? I'm not sure because I don't have some knowledge on the subject, but is it possible to combine reverse connecting and socket pairs?

Reply:
i don't think so. reverse controls how the sockfd is created, by accept() or by connect(). but when fd is used, we already have the fd. it doesn't make sense to allow reverse anymore.

definitely, i can combine the two branches into one statement using `?:`